src/Microsoft.ComponentDetection.Detectors/dotnet/DotNetComponentDetector.cs

```diff
@@ -0,0 +1,186 @@
+namespace Microsoft.ComponentDetection.Detectors.DotNet;
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using global::NuGet.ProjectModel;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+
+public class DotNetComponentDetector : FileComponentDetector, IExperimentalDetector
+{
+    private const string GlobalJsonFileName = "global.json";
+    private readonly ICommandLineInvocationService commandLineInvocationService;
+    private readonly IDirectoryUtilityService directoryUtilityService;
+    private readonly IFileUtilityService fileUtilityService;
+    private readonly IPathUtilityService pathUtilityService;
+    private readonly LockFileFormat lockFileFormat = new();
+    private readonly Dictionary<string, string?> sdkVersionCache = [];
```

Comment:
❤️ I like having this in cache, so we only do once per `project.csproj` location and once per global.json, assuming they live in different directories. If it is the same directory between those two then we should leverage the in-memory cache you created here.